ui/src/message_popup/author/edit_question_page.jsx

```diff
@@ -11,15 +11,67 @@ export default React.createClass({
   displayName: 'EditQuestionPage',
 
   propTypes: {
-    questionId: React.PropTypes.string.isRequired
+    questionId: React.PropTypes.string.isRequired,
+    loaded: React.PropTypes.bool,
+    allQuestions: React.PropTypes.object,
+    onReloadQuestions: React.PropTypes.func
+  },
+
+  componentDidMount(){
+    this.props.onReloadQuestions();
```

Comment:
I originally had that but `onReloadQuestions` would not be called after rerouting. The most important one being not reloading the questions after editing a question and returning to `QuestionsPage`. I think it's because React swaps out one `QuestionsLoader` with another identical one and it has no indication that the `QuestionsLoader` component actually changed and so it is 'already mounted'. Maybe adding keys to the `QuestionsLoader`s would fix this but I felt the most surefire way of making sure that the questions are reloading when necessary is doing so in the components that need them.